src/gate.ts

```diff
@@ -650,15 +640,11 @@ export class Gate {
         })
     }
     close() {
-        this.clear()
         this.stopBoarding()
         if (this.session) {
             this.session.close()
             this.session = null
         }
-        // TODO: this doesn't belong here
```

Comment:
Reading your code gave me an idea: add `Shell.onGateClosed` that will go home only if the gate is the active one.

Reply:
Same as above, we need a way to check if a gate is loaded or just connecting.

Reply:
I think checking for `this.activeW` is a good way to approach this.